mtp_api/apps/security/serializers.py

```diff
@@ -220,9 +224,92 @@ def update(self, instance, validated_data):
+    added_by = BasicUserSerializer(read_only=True)
+
+    class Meta:
+        model = CheckAutoAcceptRuleState
+        fields = [
+            'added_by',
+            'active',
+            'reason',
+            'created',
+            'auto_accept_rule'
+        ]
+        read_only_fields = (
+            'id',
+            'auto_accept_rule'
+        )
+
+    def validate(self, attrs):
+        if self.instance and attrs.get('active') is None:
+            raise ValidationError({
+                'active': ValidationError(_('On update, states[0].active must be specified'))
+            })
+        return super().validate(attrs)
+
+
+class CheckAutoAcceptRuleSerializer(serializers.ModelSerializer):
+    states = CheckAutoAcceptRuleStateSerializer(many=True, required=True)
+
+    def validate(self, attrs):
+        if len(attrs['states']) != 1:
+            raise ValidationError(
+                _(f'When creating or updating an auto-accept rule, states must be of length 1, not {len(attrs.states)}')
+            )
+        return super().validate(attrs)
+
+    def create(self, validated_data):
+        auto_accept_rule = CheckAutoAcceptRule.objects.create(
+            debit_card_sender_details=validated_data['debit_card_sender_details'],
+            prisoner_profile=validated_data['prisoner_profile'],
+        )
+        CheckAutoAcceptRuleStateSerializer().create(
+            validated_data={
+                'active': True,
+                'reason': validated_data['states'][0]['reason'],
+                'added_by': self.context['request'].user,
+                'auto_accept_rule': auto_accept_rule
+            }
+        )
+        auto_accept_rule.refresh_from_db()
+        return auto_accept_rule
+
+    def update(self, instance, validated_data):
+        # The only operation we support here is to create a new associated state
+        instance.states.add(
+            CheckAutoAcceptRuleStateSerializer().create(
+                validated_data={
+                    'active': validated_data['states'][0]['active'],
+                    'reason': validated_data['states'][0]['reason'],
+                    'added_by': self.context['request'].user,
+                    'auto_accept_rule': instance
+                }
+            )
+        )
+        instance.save()
+        return instance
+
+    class Meta:
+        model = CheckAutoAcceptRule
+        fields = [
+            'id',
+            'created',
+            'debit_card_sender_details',
+            'prisoner_profile',
+            'states',
+        ]
+        read_only_fields = [
+            'id',
+            'created',
+        ]
+
+
 class CheckSerializer(serializers.ModelSerializer):
     actioned_by_name = serializers.SerializerMethodField('get_actioned_by_name_from_user')
     assigned_to_name = serializers.SerializerMethodField('get_assigned_to_name_from_user')
+    auto_accept_rule_state = CheckAutoAcceptRuleStateSerializer(read_only=True, required=False)
```

Comment:
is it worth _always_ having this present? i know we need this for the auto-accept review page and we need the _latest_ state probably always. see, i'm just thinking of the credits to action and history pages which will be loading additional models for every row and then not displaying them. then again, we are not expecting a huge number of auto-accepts anyway. just thinking "out loud"

Reply:
So as I understand it the history page will eventually need this information, as it's used to determine whether a check has been auto-accepted rather than just accepted (as they are to be displayed differently). Credits to action is a fair point, I could break this out into another serializer so that it doesn't get loaded for that case, but we already have a confusing hierarchy of check serializers and I don't want to make it worse. Ideally we'd want a way of selectively choosing which relations to serialize (a company I used to work for made a library for just this use case https://django-rest-framework-serializer-extensions.readthedocs.io/en/latest/) but i'd see that as a seperate refactoring ticket(s) as there's quite a few places we could implement it